lib/cisco_node_utils/radius_server.rb

```diff
@@ -48,24 +66,70 @@ def self.radiusservers
       radiusservers_list = config_get('radius_server', 'hosts')
       return hash if radiusservers_list.empty?
       radiusservers_list.each do |id|
-        hash[id] = RadiusServer.new(id, false)
+        if platform == :ios_xr
+          authp = config_get('radius_server', 'auth-port', id)
+          authp = authp[0] if authp.is_a?(Array)
```

Comment:
Since you've flagged this as `multiple` in the YAML, it should always be an Array.